### PR TITLE
[BEAM-928] Enable Spotbugs in KafkaIO, fix warnings

### PR DESCRIPTION
**Pull request by aromanenko-dev**
Enabled Spotbugs for KafkaIO and fixed some warnings after.
One of them was related to `KafkaRecord` which I think doesn't need to be `Serializable`

R: @iemejia @rangadi (if you have a time) 
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by rangadi** [comment]
👍 LGTM.

File: sdks/java/io/kafka/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@
  */
 
 plugins { id 'org.apache.beam.module' }
-applyJavaNature(enableSpotbugs: false)
+applyJavaNature(enableSpotbugs: true)
```
Comment (iemejia) [inline sdks/java/io/kafka/build.gradle]: Please remove the variable the default is true so no need to do it twice.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java [hunk 2]
```diff
@@ -520,6 +521,7 @@ void commitTxn(long lastRecordId, Counter numTransactions) throws IOException {
       }
     }
 
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
```
Comment (iemejia) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java]: The goal of the JIRA is to fix the warnings, in this case deal with the different exceptions. Can't this be fixed?
If it is really impossible to fix, prefer using the `findbugs-filter.xml` (or `spotbugs-filter.xml`) file to do the exclusions.

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java [hunk 3]
```diff
@@ -28,7 +27,7 @@
  * KafkaRecord contains key and value of the record as well as metadata for the record (topic name,
  * partition id, and offset).
  */
-public class KafkaRecord<K, V> implements Serializable {
+public class KafkaRecord<K, V> {
```
Comment (iemejia) [inline sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java]: `KafkaRecord` should be Serializable. Or something deeply changed in the IO, Isn't this the payload of the Reader (aka what is sent in the network?)

Comment (aromanenko-dev) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java]: For payload we use coders, don't we? `KafkaRecordCoder` is used for `KafkaRecord` in this case.

Comment (iemejia) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaRecord.java]: Yes you are right this is `Coder`'s job, good point.

File: sdks/java/io/kafka/build.gradle (hunk 1 above)
Comment (aromanenko-dev) [reply sdks/java/io/kafka/build.gradle]: Thanks, fixed

File: sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java (hunk 2 above)
Comment (aromanenko-dev) [reply sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaExactlyOnceSink.java]: Yes, you are right, fixed by changing type of caught exception

**Comment by aromanenko-dev** [comment]
@iemejia Thanks, I addressed you comments, PTAL

**Pull request merged**